### ISSUE TITLE
Is there any pretrained weights?

### ISSUE THREAD

**Issue by dsj320** [junwankimm/CSFM#1]

**Comment by junwankimm** [comment]
Thanks for your interest. We plan to release the pretrained weights after the paper is accepted.

**Comment by dsj320** [comment]
@junwankimm Hi, thanks for open-sourcing this project.

I found a potential issue in the alignment loss implementation (in src/train_t2i.py), which might explain unstable/poor training quality when align_loss_type="normalized_l2".

What I observed
In the l2/normalized_l2 branch:

mu_reshaped and x1 are shaped like [B, C, H, W] (e.g. [B, 768, 16, 16])
align_loss_fn(..., normalize=True) is called
inside align_loss_fn, normalization is:
F.normalize(x, p=2, dim=-1)
F.normalize(y, p=2, dim=-1)
For 4D image-like features, dim=-1 means normalization along W only, i.e. per (b, c, h, :) vector.

Why this may be problematic
For feature alignment, it is usually more meaningful to normalize either:

along channel dimension (dim=1) at each spatial location, or
after flattening all non-batch dims.
Normalizing only along width (W) seems unintended and may provide a misaligned training signal.

Suggested fix (minimal)
If the intention is channel-wise feature alignment at each spatial location:

x = F.normalize(x, p=2, dim=1) if normalize else x
y = F.normalize(y, p=2, dim=1) if normalize else y
Alternatively, if global vector normalization is intended, flatten first and normalize on the flattened dimension.

**Comment by junwankimm** [comment]
Hi, thanks for pointing this out. You are correct that the released code does not match the intended alignment implementation. Our intended setup is to apply the alignment loss after reshaping the features to (B, 256, 768) and normalizing over dim=-1, i.e., global vector normalization over the feature dimension. The core experiments were conducted with this intended formulation, and we will update the released code to make this consistent.

The intended implementation is:

```
x1_reshaped = x1.flatten(2).transpose(1, 2)
normalize = align_loss_type == "normalized_l2"
align_loss = align_loss_fn(mu, x1_reshaped, normalize=normalize)
```

Thank you again for your careful reading and interest in our work.
